### DISCUSSION TITLE
Disable JavaScript for Robots

### DISCUSSION THREAD

**Discussion by xHomu** [epicweb-dev/epic-stack#144] [Ideas]
We can greatly improve synthetic benchmark scores (Lighthouse, WebPageTest, PageSpeed) by not shipping JavaScript bundles whenever a bot crawler is detected. 

## Disabling <Script />

Since it has no effect on actual end-user experience, this is a bit of a cheat:

* Reference: https://sergiodxa.com/articles/improve-seo-by-not-sending-js-in-remix
* Implementation Sample https://github.com/manawiki/core/commit/a6628df30a458b93c2a017b8e866563d172c576b

However, most robots cannot employ client-side JS in the first place, so Remix apps tends to be unfairly penalized for the performance boost it would otherwise provide to regular users. I find the practice somewhat justifiable.

### Before/After

The impact can be quite dramatic! 

![Before/After](https://github.com/epicweb-dev/epic-stack/assets/84349818/ac16f351-8c1c-433a-9fff-26b905ac4b38)

* **[Filmstrip comparison](https://www.webpagetest.org/video/compare.php?tests=230531_BiDc64_CVY,230531_BiDcTM_DK4)**

## Concerns

* A naïve implementation may impact testing.
* Some sites that dependent on client JS might become difficult to crawl (although this would be an issue regardless for bots).

**Comment by kentcdodds** [comment]
I'm not interested in chasing the numbers for the sake of it. I don't want to change anything in the app just for making these numbers look better. If it doesn't actually improve the user experience then I'm not interested.

Thanks though!
